### ISSUE TITLE
Some chained calls are mistakenly marked as not covered

### ISSUE THREAD

**Issue by vberger** [xd009642/tarpaulin#49]
I have a very small project https://github.com/vberger/token_store which should achieve full coverage, though it does not: in some chained method calls, arranged over multiple lines by rustfmt, some of the lines are marked as uncovered, even though the final return value of the chaining is used.

Similarly, instanciations of `PhantomData` in struct initialization are marked as not covered.

See this coverage for examples of both: https://coveralls.io/builds/13651879/source?filename=src%2Flib.rs

**Comment by xd009642** [comment]
Thanks for this and a nice small example to run it on. This one might be a bit tricky, but I'll keep you posted!

**Comment by xd009642** [comment]
So, this ones been a long time coming. But I've sorted the `PhantomData` issue and most of the chained calls are sorted. There still seems to be an issue with some chained calls but from what I can see they're all collect statements with the question mark operator a la `collect::<_>()?`. As other collect calls don't register as coverable this may be the early return being instrumented instead. 

I am going to close this issue and move any updates on these new things I've found to #63 as always latest and greatest changes can be tried out from the develop branch :smile:
